src/main/java/com/ot/restaurant/repositories/CustomerRepositoryImpl.java

```diff
@@ -0,0 +1,81 @@
+package com.ot.restaurant.repositories;
+
+import com.ot.restaurant.entities.Customer;
+import com.ot.restaurant.exceptions.CustomerIdNullPointerException;
+import com.ot.restaurant.exceptions.CustomerStatusNotActiveException;
+import constants.Status;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class CustomerRepositoryImpl implements CustomerRepository {
+  private final List<Customer> customerList = new ArrayList<>();
+  private Long nextId = 0L;
+
+  private Long generateNextId() {
+    Long currentId = nextId;
+    nextId++;
+    return currentId;
+  }
+
+  public List<Customer> findAll() {
+    return customerList.stream()
+        .filter(customer -> customer.getStatus().equals(Status.ACTIVE))
+        .toList();
+  }
+
+  public Optional<Customer> findById(Long id, Status status)
+          throws Exception {
+    if (status == null) {
+      status = Status.ACTIVE;
+    }
+    if (id == null){return (Optional<Customer>) Optional.empty().orElseThrow(CustomerIdNullPointerException::new);}
+
+    return Optional.ofNullable((customerList.get(Math.toIntExact(id))));
```

Comment:
`id` might not match the index of the location of the object, don't rely on this.